src/test/regress/sql/multi_insert_select.sql

```diff
@@ -606,3 +606,34 @@ SET client_min_messages TO INFO;
 -- Views does not work
 CREATE VIEW test_view AS SELECT * FROM raw_events_first;
 INSERT INTO raw_events_second SELECT * FROM test_view;
+
+-- we need this in our next test
+truncate raw_events_first;
+
+SET client_min_messages TO DEBUG4;
+
+-- first show that the query works now
+INSERT INTO raw_events_first SELECT * FROM raw_events_second;
+
+SET client_min_messages TO INFO;
+truncate raw_events_first;
+SET client_min_messages TO DEBUG4;
+
+-- now show that it workrs for a single shard query as well
+INSERT INTO raw_events_first SELECT * FROM raw_events_second WHERE user_id = 5;
+
+SET client_min_messages TO INFO;
+
+-- if a single shard of the SELECT is unhealty, the query should fail
+UPDATE pg_dist_shard_placement SET shardstate = 3 WHERE shardid = 13300004 AND nodeport = :worker_1_port;
+truncate raw_events_first;
+SET client_min_messages TO DEBUG4;
+
+-- this should fail
+INSERT INTO raw_events_first SELECT * FROM raw_events_second;
+
+-- this should also fail
+INSERT INTO raw_events_first SELECT * FROM raw_events_second WHERE user_id = 5;
+
+-- but this should work given that it hits different shard
+INSERT INTO raw_events_first SELECT * FROM raw_events_second WHERE user_id = 6;
```

Comment:
Ok, manual with testing I showed that it works correctly. However, I found another bug on the executor side, which ends up hitting `if (currentAffectedTupleCount != previousAffectedTupleCount)` check.
So, I had to fix that as well for adding this regression test. 

Do you think is that OK?